app/controllers/admin/application_pets_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::ApplicationPetsController < ApplicationController
+
+  def update
```

Comment:
Awesome job routing the create and update actions for adopting/approving pets to a joins controller!!

This is minor but for this controller action, even though it's triggered from an admin page, it would've been okay to put this update action in the other ApplicationPetsController rather than make a separate admin joins controller. Because there's no requirement  on the path for this update including `/admin`, there's only a requirement that the show page that it redirects back to includes `/admin`. This update action is sort of a `behind-the-scenes` action similar to the create action in the joins controller. It doesn't render a view that the user would see, but instead just redirects back to the admin dashboard. So, it's not strictly required that it be an `admin` namespaced controller. Let me know if this doesn't make sense and I'm happy to chat further.